src/Shared/USConverters/CreditQualityConverter.ts

```diff
@@ -0,0 +1,101 @@
+/* *
+ *
+ *  (c) 2009-2025 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Jedrzej Ruta
+ *
+ * */
+
+
+'use strict';
+
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import MorningstarConverter from '../MorningstarConverter';
+import type { XRayUSConverterOptions } from '../../XRayUS/XRayUSOptions';
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+
+export class CreditQualityConverter extends MorningstarConverter {
+
+
+    /* *
+     *
+     *  Constructor
+     *
+     * */
+
+
+    public constructor (
+        options?: XRayUSConverterOptions
+    ) {
+        super(options);
+    }
+
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+
+    public override parse (
+        options: XRayUSConverterOptions,
+        hasMultiple?: boolean
+    ): void {
+        const table = this.table,
+            columnSuffix = hasMultiple ? `_${options.json.PortfolioName}` : '',
+            portfolio =
+                options.json.Analysis.FixedIncomeAnalysis.CreditQuality.Portfolio,
+            benchmark = options.json.Analysis.FixedIncomeAnalysis.CreditQuality.Benchmark;
+
+        if (portfolio) {
+            let rowIndex = 0;
+
+            for (const [key, value] of Object.entries(portfolio)) {
+                table.setCell('Type', rowIndex, key);
+                table.setCell('Value' + columnSuffix, rowIndex, value);
+
+                ++rowIndex;
+            }
+        }
+
+        if (benchmark) {
+            let rowIndex = 0;
+
+            for (const [key, value] of Object.entries(benchmark)) {
+                table.setCell('Type', rowIndex, key);
+                table.setCell('Value_Benchmark' + columnSuffix, rowIndex, value);
+
+                ++rowIndex;
+            }
+        }
```

Comment:
Once you change the types to more explicit ones instead of `Record` you won't be able to use Object.entires, since by default `value` is a type of any in TS, you can use the approach that I have used here https://github.com/highcharts/connectors-morningstar/pull/120/commits/9107ad20e0361f6dcbabe7b83b6273fc7cc5db83 instead.